examples/TransactionGenerator.hs

```diff
@@ -48,246 +66,484 @@ import Servant.API
-      , "timingDistribution" .= _timingDistribution o
+      [ "scriptCommand"       .= _scriptCommand o
+      , "nodeChainId"         .= _nodeChainId o
+      , "isChainweb"          .= _isChainweb o
+      , "chainwebHostAddress" .= _chainwebHostAddress o
+      , "chainwebVersion"     .= _nodeVersion o
       ]
 
-instance FromJSON (TransactionConfig -> TransactionConfig) where
-  parseJSON = withObject "TransactionConfig" $ \o -> id
-    <$< scriptCommand ..: "scriptCommand" % o
-    <*< nodeChainId ..: "nodeChainId" % o
-    <*< serverRootPrefix ..: "serverRootPrefix" % o
-    <*< isChainweb ..: "isChainweb" % o
-    <*< chainwebNodePort ..: "chainwebNodePort" % o
-    <*< timingDistribution ..: "timingDistribution" % o
-
-data GeneratorState s = GeneratorState
-  { _gsGen :: Gen s
+instance FromJSON (ScriptConfig -> ScriptConfig) where
+  parseJSON = withObject "ScriptConfig" $ \o -> id
+    <$< scriptCommand       ..: "scriptCommand"       % o
+    <*< nodeChainId         ..: "nodeChainId"         % o
+    <*< isChainweb          ..: "isChainweb"          % o
+    <*< chainwebHostAddress ..: "chainwebHostAddress" % o
+    <*< nodeVersion         ..: "chainwebVersion"     % o
+    <*< logHandleConfig     ..: "logging"             % o
+
+data ServantRecord = ServantRecord
+  { apiSend   :: SubmitBatch     -> ClientM RequestKeys
+  , apiPoll   :: Poll            -> ClientM PollResponses
+  , apiListen :: ListenerRequest -> ClientM ApiResult
+  , apiLocal  :: Command Text    -> ClientM (CommandSuccess Value)
+  }
+
+
+data TransactionGeneratorState = TransactionGeneratorState
+  { _gsGen     :: Gen (PrimState IO)
   , _gsCounter :: Int64
   }
 
-makeLenses ''GeneratorState
-
-defaultTransactionConfig :: TransactionConfig
-defaultTransactionConfig =
-  TransactionConfig
-    { _scriptCommand      = DeployContracts []
-    , _nodeChainId        = unsafeChainId 0
-    , _serverRootPrefix   = "https://us1.chainweb.com"
-    , _isChainweb         = True
-    , _chainwebNodePort   = ChainwebPort 443
-    , _timingDistribution = def
+makeLenses ''TransactionGeneratorState
+
+defaultScriptConfig :: ScriptConfig
+defaultScriptConfig =
+  ScriptConfig
+    {
+      _scriptCommand       = RunSimpleExpressions def def
+    , _nodeChainId         = unsafeChainId 0
+    , _isChainweb          = True
+    , _chainwebHostAddress = HostAddress (unsafeHostnameFromText "127.0.0.1") (unsafePortFromText "1789")
+    -- , _nodeVersion      = "testnet00"
+    , _nodeVersion         = fromJuste $ chainwebVersionFromText "testWithTime"
+    , _logHandleConfig     = U.StdOut
     }
 
-transactionConfigParser :: MParser TransactionConfig
-transactionConfigParser = id
-  <$< scriptCommand .:: option auto
-      % long "script-command" <> short 'c'
-      <> help "The specific command to run (DeployContracts|RunStandardContracts|RunSimpleExpressions)."
-  <*< nodeChainId .::option auto
+scriptConfigParser :: MParser ScriptConfig
```

Comment:
Could this stuff be used to do script parsing instead of the above manual parsing?